src/multiSceneMovie.coffee

```diff
@@ -28,23 +29,40 @@ class MultiSceneMovie
   composeScenes = (rootElement, movie, movieScene, movieLoadScene, movieFinishScene, contentScene) ->
     ee = new EventEmitter(rootElement)
 
-    bindTrackingEvent = (event, url, type)->
-      switch event
-        when "creative_view"
-          ee.listen("movie:screen:appeared", -> new Beacon(url, type))
-        when "start"
-          ee.listen("movie:started", -> new Beacon(url, type))
-        when "first_quartile"
-          ee.listen("movie:played:firstQuartile", -> new Beacon(url, type))
-        when "mid_point"
-          ee.listen("movie:played:half", -> new Beacon(url, type))
-        when "third_quartile"
-          ee.listen("movie:played:thirdQuartile", -> new Beacon(url, type))
-        when "complete"
-          ee.listen("movie:finished", -> new Beacon(url, type))
-
-    for tracking in rootElement.getElementsByClassName("tracking")
-      bindTrackingEvent(tracking.dataset.event, tracking.dataset.url, tracking.dataset.type)
+    composeVastTrackingEvents = (eventEmitter) ->
+      bindTrackingEvent = (eventEmitter, tracking)->
+        return unless eventEmitter
+        switch tracking.event
+          when "creative_view"
+            eventEmitter.listen("movie:screen:appeared", -> new Beacon(tracking.url, tracking.type))
```

Comment:
[nit] Alternatively, you could just define the event name in the switch statement to save some typing.

e.g.

```coffee
switch tracking.event
  when "creative_view"
    eventName = "movie:screen:appeared"

eventEmitter.listen(eventName, -> new Beacon(tracking.url, tracking.type))
```